### DISCUSSION TITLE
Client vs Server Components - Data Fetching

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#49947] [Help]
### Summary

Just a general clarification as I am trying to better understand the separation and migration some web site apps from an older Nextjs version to the newer model.

In the doc, I get the example of thinking through Client vs Server Components: https://nextjs.org/docs/getting-started/react-essentials.  But even at the top of the page where it shows an example application that has Server/Client Components, it calls out "Search" as a Client Component (which is what I suspect).  But later in the article and in frontendmasters nextjs 13 course, the recommendation is that you do not do data fetching from a Client Component, even though a Search has to make api calls with search query data and retrieve and provide the results, including client-side pagination (API calls).

So I am trying to understand, because this is specifically our use case where there is intentional CSR that requires user interaction and API calls for the data.  Is this an anti-pattern then to put the data fetching into the Client Component?  And maybe I am misunderstanding the concept but I assume the Server Component is fully compiled into HTML/React and then cached on the CDN, therefore I need CSR for this interaction, which is where I assume Client Components come into play (like we are using regular React Components today).

**Comment by devjiwonchoi** [comment]
Note that this is my personal opinion. Data management is indeed recommended to be worked under the server side, but the search method seems to be recommended and supported on the client side.

I assume that data fetching from the server side is most recommended for dynamic routing and SEO purposes.
In the case of the search method, Next13 App Router presents[ useSearchParams](https://nextjs.org/docs/app/api-reference/functions/use-search-params) and is guiding to use Client Component for search querying. 

It is because the search method does not require generating static pages or SEO and so forth, which are one of the main purposes of using server-side fetching. Therefore the modification of the components are the only factors required for displaying search results, and I guess this is why it is recommended to use as Client Component.

**Reply by jaybytez** [reply]
@devjiwonchoi , thanks for that explanation.  That makes total sense to me and seems like the way I interpret the separation as well.

**Reply by devjiwonchoi** [reply]
Glad that my answer was helpful. Happy Coding!
